iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -871,7 +871,11 @@ PortManager::acquirePublisherPortDataWithoutDiscovery(const capro::ServiceDescri
         return cxx::error<PortPoolError>(PortPoolError::UNIQUE_PUBLISHER_PORT_ALREADY_EXISTS);
     }
 
-    if (isInternal(service) && runtimeName != RuntimeName_t{IPC_CHANNEL_ROUDI_NAME})
+    if (runtimeName == RuntimeName_t{IPC_CHANNEL_ROUDI_NAME})
+    {
+        m_internalServices.push_back(service);
+    }
+    else if (isInternal(service))
```

Comment:
Isn't this dangerous? Can't this be done by an optional parameter (e.g. internalPort) which is by default false and the `addInternal...` call this method with `true`?